### PR TITLE
LWM2M: Add sol_lwm2m_payload type.

### PR DESCRIPTION
**Pull request by cabelitos**
This struct will make things easier when implementing the access control.

Signed-off-by: Guilherme Iscaro <guilherme.iscaro@intel.com>

### CODE REVIEW

**Comment by cabelitos** [comment]
@bsmelo Take a look in this commit

File: src/lib/comms/include/sol-lwm2m.h [hunk 1]
```diff
@@ -394,6 +394,28 @@ struct sol_lwm2m_resource {
...
+ * @see struct sol_lwm2m_object::create
+ */
+struct sol_lwm2m_payload {
+    /** @brief The payload type
+     * @see sol_lwm2m_content_type
+     */
+    enum sol_lwm2m_content_type type;
+    union sol_lwm2m_payload_data {
+        /** @brief The payload content as TLV format -
+            use only when @c sol_lwm2m_payload::type is #SOL_LWM2M_CONTENT_TYPE_TLV
```
Comment (bsmelo) [inline src/lib/comms/include/sol-lwm2m.h]: This line is missing the initial `*`. I don't know if it's mandatory for doxygen though.

File: src/lib/comms/sol-lwm2m.c [hunk 2]
```diff
@@ -3013,7 +3003,15 @@ handle_resource(void *data, struct sol_coap_server *server,
...
 
     if (r < 0)
-        content_format = SOL_LWM2M_CONTENT_TYPE_TEXT;
+        payload.type = SOL_LWM2M_CONTENT_TYPE_TEXT;
+    else
+        payload.type = content_format;
+
+    if (payload.type == SOL_LWM2M_CONTENT_TYPE_JSON) {
+        SOL_WRN("JSON content format is not supported");
+        header_code = SOL_COAP_RESPONSE_CODE_BAD_REQUEST;
```
Comment (bsmelo) [inline src/lib/comms/sol-lwm2m.c]: According to Pg. 74 of the spec you should actually throw 4.15 Unsupported content format (`SOL_COAP_RESPONSE_CODE_UNSUPPORTED_CONTENT_FORMAT`).

Comment (cabelitos) [inline src/lib/comms/sol-lwm2m.c]: I will fix it,

**Comment by cabelitos** [comment]
merged

**Pull request closed without merging**